woocommerce-gateway-stripe.php

```diff
@@ -225,6 +225,52 @@ public function install() {
 
 					add_woocommerce_inbox_variant();
 					$this->update_plugin_version();
+
+					// TODO: Remove this when we're reasonably sure most merchants have had their
+					// settings updated like this. ~80% of merchants is a good threshold.
+					// - @reykjalin
+					$this->update_prb_location_settings();
+				}
+			}
+
+			/**
+			 * Updates the PRB location settings based on deprecated filters.
+			 *
+			 * The filters were removed in favor of plugin settings. This function can, and should,
+			 * be removed when we're reasonably sure most merchants have had their settings updated
+			 * through this function. Maybe ~80% of merchants is a good threshold?
+			 *
+			 * @since x.x.x
+			 * @version x.x.x
+			 */
+			public function update_prb_location_settings() {
+				$stripe_settings = get_option( 'woocommerce_stripe_settings', [] );
+				$prb_locations   = isset( $stripe_settings['payment_request_button_locations'] )
+					? $stripe_settings['payment_request_button_locations']
+					: [];
+				if ( ! empty( $stripe_settings ) && empty( $prb_locations ) ) {
```

Comment:
```php
if ( empty( $prb_locations ) ) {
```

I think just cheking on `$prb_locations` is enough here?

Reply:
I'm not so sure 🤔 

If there are no plugin settings available, i.e. `$stripe_settings` is empty, then `empty( $prb_locations )` would evaluate to `true`, whereas `! empty( $stripe_settings ) && empty( $prb_locations )` would evaluate to `false`.

I'm not sure what implications it has to have the `$prb_locations` be the only thing written to the stripe settings, but I figured it'd be better to fall back on default values if there are no Stripe plugin settings available.

What do you think?

Reply:
Thanks for the explanation. I figured out why you are checking on `$stripe_settings` too. You're right!